dotnet-desktop-guide/net/winforms/controls-design/custom-controls-overview.md

```diff
@@ -0,0 +1,166 @@
+
+## Base class
+
+There are two base classes to choose from when creating a custom control:
+
+- <xref:System.Windows.Forms.Control?displayProperty=fullName>
+
+  This is the same base class used by other Windows Forms controls. You control the input and output of the control directly.
+
+- <xref:System.Windows.Forms.ScrollableControl?displayProperty=fullName>
+
+  Some Windows Forms controls use this base class. This class extends `Control` by adding the ability to scroll the content.
+
+Unless you require scrolling the contents of the custom control, use `Control` as your base class.
+
+## Inherited capabilities
+
+Since the base class of a custom control is <xref:System.Windows.Forms.Control>, you automatically inherit Windows Forms functionality shared by all controls. Here are a few of the capabilities you get with a custom control:
+
+- Keyboard a mouse input.
+- Layout behaviors, such as anchoring and docking.
+- Support for tabbing.
+- Minimum and maximum size restrictions.
+
+## Painting
+
+<!-- I don't like the second half of this para -->
+Painting, which means to draw the control's visual, is accomplished by overriding the <xref:System.Windows.Forms.Control.OnPaint%2A> method. For more information about how painting is used by controls, see [Painting and drawing on controls](../controls/custom-painting-drawing.md).
+
+When you create a custom control using the Visual Studio templates, the `OnPaint` method is automatically overridden for you. The template does this because you're required to write the code to draw your control. Here's an example of what the template generates:
+
+```csharp
+public partial class CustomControl1 : Control
+{
+    public CustomControl1()
+    {
+        InitializeComponent();
+    }
+
+    protected override void OnPaint(PaintEventArgs pe)
+    {
+        base.OnPaint(pe);
+    }
+}
+```
+
+```vb
+Public Class CustomControl1
+
+    Protected Overrides Sub OnPaint(ByVal e As System.Windows.Forms.PaintEventArgs)
+        MyBase.OnPaint(e)
+
+        'Add your custom paint code here
+    End Sub
+
+End Class
+```
+
+A custom control is painted with the <xref:System.Windows.Forms.Control.OnPaint%2A> method. The single argument of this method is a <xref:System.Windows.Forms.PaintEventArgs> object, which provides all of the information and functionality required to render your control. `PaintEventArgs` provides two properties that are used in rendering your control:
+
+- <xref:System.Windows.Forms.PaintEventArgs.ClipRectangle%2A?displayProperty=nameWithType>&mdash;Represents the part of the control needs to be redrawn. This can be the entire control or part of the control.
```

Comment:
```suggestion
- <xref:System.Windows.Forms.PaintEventArgs.ClipRectangle%2A?displayProperty=nameWithType>&mdash;Represents the part of the control that needs to be redrawn. This can be the entire control or part of the control.
```